src/core/storage/client.ts

```diff
@@ -0,0 +1,168 @@
+import { StorageClient, StorageConfig, UploadResult, RetrieveResult, UploadOptions, UploadFile } from './types.js';
+import { Signer } from '@ucanto/principal/ed25519';
+import { StoreMemory } from '@web3-storage/w3up-client/stores/memory';
+import * as Storage from '@web3-storage/w3up-client';
+import { parseDelegation } from './utils.js';
+import { DEFAULT_GATEWAY_URL } from './config.js';
+
+/**
+ * Implementation of the StorageClient interface for Storacha network
+ */
+export class StorachaClient implements StorageClient {
+  private config: StorageConfig;
+  private initialized: boolean = false;
+  private storage: Storage.Client | null = null;
+
+  constructor(config: StorageConfig) {
+    this.config = {
+      ...config,
+      gatewayUrl: config.gatewayUrl?.trim() || DEFAULT_GATEWAY_URL
+    };
+  }
+
+  /**
+   * Initialize the storage client and establish connection based on the storage config
+   */
+  async initialize(): Promise<void> {
+    if (this.initialized) {
+      return;
+    }
+
+    if (!this.config.privateKey) {
+      throw new Error('Private key is required');
+    }
+
+    if (!this.config.delegation) {
+      throw new Error('Delegation is required');
+    }
+
+    try {
+      const principal = Signer.parse(this.config.privateKey);
+      const store = new StoreMemory();
+      this.storage = await Storage.create({ principal, store });
+
+      const delegationProof = await parseDelegation(this.config.delegation);
+      const space = await this.storage.addSpace(delegationProof);
+      await this.storage.setCurrentSpace(space.did());
```

Comment:
I think this is redundant because you're using a fresh memory store that has no existing spaces.

Reply:
Fixed.